api/v1alpha2/azuremachine_conversion.go

```diff
@@ -81,6 +86,13 @@ func Convert_v1alpha3_AzureMachineSpec_To_v1alpha2_AzureMachineSpec(in *infrav1a
 		return err
 	}
 
+	if in.FailureDomain != nil {
+		out.AvailabilityZone = AvailabilityZone{
```

Comment:
Should this first test that it isn't overriding an existing value for out.AvailabilityZone first?

Reply:
I've added a change to do this